tools/kraken.wdl

```diff
@@ -0,0 +1,336 @@
+## # Kraken2
+##
+## Methods for bootstrapping and running [Kraken2](https://github.com/DerrickWood/kraken2)
+
+version 1.0
+
+task download_taxonomy {
+    input {
+        String db_name = "kraken2_taxonomy"
+        Int memory_gb = 4
+        Int disk_size_gb = 60
+        Int max_retries = 3
+    }
+
+    parameter_meta {
+        db_name: "Name for output tarball. The suffix `.tar.gz` will be added."
+        memory_gb: "RAM to allocate for task"
+        disk_size_gb: "Disk space to allocate for task"
+        max_retries: "Number of times to retry in case of failure"
+    }
+
+    command <<<
+        set -euo pipefail
+
+        kraken2-build --download-taxonomy \
+            --use-ftp \
+            --db ~{db_name} 2>&1 \
+            | awk '/gunzip:/ { print; exit 42 } !/gunzip:/ { print }' 1>&2
+
+        tar -C ~{db_name}/ -czf "~{db_name}.tar.gz" .
+
+        rm -r ~{db_name}
+    >>>
+ 
+    runtime {
+        memory: memory_gb + " GB"
+        disk: disk_size_gb + " GB"
+        cpu: 1
+        docker: 'ghcr.io/stjudecloud/kraken2:branch-kraken2-1.0.0'
+        maxRetries: max_retries
+    }
+
+    output {
+        File taxonomy = db_name + ".tar.gz"
+    }
+
+    meta {
+        author: "Andrew Frantz"
+        email: "andrew.frantz@stjude.org"
+        description: "This WDL task downloads the Kraken2 taxonomy."
+    }
+}
+
+task download_library {
+    input {
+        String library_name
+        String db_name = "kraken2_"+library_name+"_library"
+        Int memory_gb = 4
+        Int added_disk_size_gb = 0
+        Int max_retries = 3
+    }
+
+    parameter_meta {
+        library_name: "One of `archaea`, `bacteria`, `plasmid`, `viral`, `human`, `fungi`, `plant`, `protozoa`, `nt`, `UniVec`, or `UniVec_Core`"
+        db_name: "Name for output tarball. The suffix `.tar.gz` will be added."
+        memory_gb: "RAM to allocate for task"
+        added_disk_size_gb: "Additional disk space to allocate for task. Default disk size is determined dynamically based on `library_name`."
+        max_retries: "Number of times to retry in case of failure"
+    }
+
+    Int disk_size_gb = (if library_name=="bacteria" then 300 else 20) + added_disk_size_gb
```

Comment:
I say in the meta that `nt` is a valid option to this task, but if `disk` gets used by the backend, this won't have enough disk space allocated. I believe `nt` requires over 2TB of space (over 1TB for the uncompressed library, library gets duplicated during repeat masking).

Should we support it in this clause? I think anybody running on a backend that allocates and charges for disk space would want to avoid using `nt`. Seems like a use case we could ignore? But that is a tad sloppy...

Reply:
I would just note in the parameter meta where you mention `nt` that you would need a corresponding change to `added_disk_size_gb` to account for `nt`. Then it is up to the user to figure out how much to use and they should understand the charges when requesting that much disk.